pkg/operator/ceph/cluster/crash/reconcile.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crash
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mgr"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
+	"github.com/rook/rook/pkg/operator/ceph/cluster/rbd"
+	"github.com/rook/rook/pkg/operator/ceph/file/mds"
+	"github.com/rook/rook/pkg/operator/ceph/object"
+
+	"github.com/coreos/pkg/capnslog"
+
+	"github.com/rook/rook/pkg/operator/ceph/cluster/osd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	logger = capnslog.NewPackageLogger("github.com/rook/rook", controllerName)
+	// Implement reconcile.Reconciler so the controller can reconcile objects
+	_ reconcile.Reconciler = &ReconcileNode{}
+)
+
+// ReconcileNode reconciles ReplicaSets
+type ReconcileNode struct {
+	// client can be used to retrieve objects from the APIServer.
+	scheme *runtime.Scheme
+	client client.Client
+}
+
+// Reconcile reconciles a node and ensures that it has a drain-detection deployment
```

Comment:
What is the purpose of looking for drain-detection here. I have to admit I'm a little confused, and I think I'm missing some k8s knowledge around this that would be relevant context.

Reply:
Oops, leftover. Nevermind.